src/routes/post_report.js

```diff
@@ -1,28 +1,43 @@
 const connPool = require('../../database/db_connect');
 const dbQueries = require('../db_queries');
+const async = require('../async.js');
 
 module.exports = {
   method: 'POST',
   path: '/post-report',
   handler: (request, reply) => {
-    const postData = JSON.parse(request.payload);
+    const formData = JSON.parse(request.payload);
 
-    dbQueries.postReportDetails(connPool, postData, (err, result) => {
-      if (err) return console.log(err);
+    // insert report details
+    const insertDetails = (postData, callback) => {
+      const insertData = postData;
+      return dbQueries.postReportDetails(connPool, insertData, (postReportDetailsError, result) => {
```

Comment:
I think one of the advantages of using waterfall is avoiding nesting, which means you can just use `error` here instead of `postReportDetailsError` - I think Rory only used more detailed names because of the nesting issue